packages/react-focus/src/components/FocusZone/FocusZone.tsx

```diff
@@ -469,6 +463,12 @@ export class FocusZone extends React.Component<IFocusZoneProps> implements IFocu
     if (stopFocusPropagation || doNotAllowFocusEventToPropagate) {
       ev.stopPropagation();
     }
+
+    if (onFocus) {
```

Comment:
consider making a test case which exercises the issue so that we don't break this.